### ISSUE TITLE
Adding mpris-services with castlabs electron v8.5.2-wvvmp results in a 403

### ISSUE THREAD

**Issue by Mastermindzh** [castlabs/electron-releases#70]
It looks like node-gyp is attempting to download the "electronjs" variant of 8.5.2-wvvmp.
Any way I can fix this?

Full details:
https://github.com/Mastermindzh/tidal-hifi/runs/1204961368?check_suite_focus=true

Isolated error:
```
node-gyp rebuild


                    errorOut=gyp WARN install got an error, rolling back install
    gyp ERR! configure error 
    gyp ERR! stack Error: 403 response downloading electronjs.org/headers/v8.5.2-wvvmp/node-v8.5.2-wvvmp.tar.gz
    gyp ERR! stack     at Request.<anonymous> (/opt/hostedtoolcache/node/12.18.4/x64/lib/node_modules/npm/node_modules/node-gyp/lib/install.js:188:14)
    gyp ERR! stack     at Request.emit (events.js:327:22)
    gyp ERR! stack     at Request.onRequestResponse (/opt/hostedtoolcache/node/12.18.4/x64/lib/node_modules/npm/node_modules/request/request.js:1066:10)
    gyp ERR! stack     at ClientRequest.emit (events.js:315:20)
    gyp ERR! stack     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:596:27)
    gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:119:17)
    gyp ERR! stack     at TLSSocket.socketOnData (_http_client.js:469:22)
    gyp ERR! stack     at TLSSocket.emit (events.js:315:20)
    gyp ERR! stack     at addChunk (_stream_readable.js:295:12)
    gyp ERR! stack     at readableAddChunk (_stream_readable.js:271:9)
    gyp ERR! System Linux 5.4.0-1025-azure
    gyp ERR! command "/opt/hostedtoolcache/node/12.18.4/x64/bin/node" "/opt/hostedtoolcache/node/12.18.4/x64/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
    gyp ERR! cwd /home/runner/work/tidal-hifi/tidal-hifi/node_modules/abstract-socket
    gyp ERR! node -v v12.18.4
    gyp ERR! node-gyp -v v5.1.0
    gyp ERR! not ok 
    npm ERR! code ELIFECYCLE
    npm ERR! errno 1
    npm ERR! abstract-socket@2.1.1 install: `node-gyp rebuild`
    npm ERR! Exit status 1
```

**Comment by khwaaj** [comment]
For general use with `electron-builder` you can see how to configure it [here](https://github.com/castlabs/electron-releases/wiki/FAQ#how-can-i-use-electron-builder-with-ecs), in case you haven't already.

In this case it seems you require the `node` package as well as the electron package though, is that right? We are not publishing those binaries in our releases since they are the same as for stock electron. I guess you'd need to find some way to configure `node-gyp` to download from the corresponding stock electron release, but I'm not familiar with `node-gyp` so I'm not sure what kind of configuration is available there.

**Comment by ArmelChesnais** [comment]
So I ran into the same issue, as I needed to package some native dependencies along with electron-wvvmp.
For anyone looking for a solution, here's how I resolved this. @Mastermindzh if you're still needing this, hope it helps you.

in you project's root package.json, set it as such.
`
"build": {
    "electronVersion": "<electron version #>",
    "electronDownload": {
      "version": "<electron version #>-wvvmp",
      "mirror": "https://github.com/castlabs/electron-releases/releases/download/v"
    },
}
`
build.electronVersion will tell Builder what version of electron to use as reference, which means it will use this value for fetching the headers.
build.electronDownload.version and build.electronDownload.mirror tells it what version and path to use to download the Electron release, which it will use to get the Electron for Content Security release with the proper ECS version from castlabs
so for example, if you're trying to use castlabs' fork of electron v10.1.2:
`
"build": {
    "electronVersion": "10.1.2",
    "electronDownload": {
      "version": "10.1.2-wvvmp",
      "mirror": "https://github.com/castlabs/electron-releases/releases/download/v"
    },
}
`

Warning: it does mean you will have to manually update these versions every time you want to update your electron version, and you will no longer benefit from builder determining the value automatically. Make sure you are matching the version actually installed in your node module/yarn.lock, not just the one listed in package.json .

@khwaaj Not sure if this is something that you might want to add to your documentation, but it would help any other users trying to add native dependencies to their ECS project.

**Comment by Mastermindzh** [comment]
@ArmelChesnais 

Because I'm using build files I had to make changes to my build files instead.
I tried the electronDownload.version and electronDownload.mirror before but didn't have the overall version set... seems it needs that.

Many thanks. Seems to work now.

These are the changes I had to do:

https://github.com/Mastermindzh/tidal-hifi/blob/feature/mpris-403-fix/build/electron-builder.yml#L2-L5
